### PR TITLE
Write server and client

### PR DESCRIPTION
**Pull request by astanishevskyi**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+		reader := bufio.NewReader(os.Stdin)
+		text, _ := reader.ReadString('\n')
+		conn.Write([]byte(text + "\n"))
+		message, err := bufio.NewReader(conn).ReadString('\n')
+		if err != nil {
+			fmt.Println(err)
+			return
+		}
+		fmt.Print("->" + message)
+	}
```
Comment (dmtbrk) [inline client.go]: Here are some suggestions:

```suggestion
	// Readers and writers are created here only once.
	inputReader := bufio.NewReader(os.Stdin)
	connReader := bufio.NewReader(conn)
	connWriter := bufio.NewWriter(conn)
	
	for {
		input, err := inputReader.ReadBytes('\n') // ReadBytes eliminates the need of conversion to bytes. 
		if err != nil {
			fmt.Println(err)
			return
		}
		
		// Using a buffered writer here is optional. Just for example.
		connWriter.Write(input)
		connWriter.WriteByte('\n')
		if err := connWriter.Flush(); err != nil {
			fmt.Println(err)
			return
		}
		
		message, err := connReader.ReadString('\n')
		if err != nil {
			fmt.Println(err)
			return
		}
		fmt.Print("->", message) // The same result, but without string concatination.
	}
```

File: server.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		switch command {
+		case "TIME":
+			conn.Write([]byte(time.Now().String() + "\n"))
+		case "REVERSE":
+			conn.Write([]byte(reverseString(text) + "\n"))
+		default:
+			conn.Write([]byte(strings.ToUpper(message) + "\n"))
+		}
+
+		conn.SetDeadline(time.Now().Add(time.Minute)) // update timeout
```
Comment (dmtbrk) [inline server.go]: Setting a deadline in two different places makes it difficult to change.

File: server.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	defer conn.Close()
+	for {
+		message, err := bufio.NewReader(conn).ReadString('\n')
+		if err != nil {
+			fmt.Println(err)
+			return
+		}
+		fmt.Print("Message received:", message)
+		splitedMessage := strings.Fields(message)
+		command := splitedMessage[0]
```
Comment (dmtbrk) [inline server.go]: Potential panic if `message` has no words.

**Review by dmtbrk — COMMENTED** [review]
Good job

File: server.go (hunk 2 above)
Comment (astanishevskyi) [reply server.go]: Fixed

File: server.go (hunk 3 above)
Comment (astanishevskyi) [reply server.go]: Fixed

**Pull request merged**